crates/core/tedge_api/Cargo.toml

```diff
@@ -16,6 +16,7 @@ download = { workspace = true }
 json-writer = { workspace = true }
 log = { workspace = true }
 mqtt_channel = { workspace = true }
+nix = { workspace = true }
```

Comment:
Changes like this makes me feel that we're bloating the `tedge-api` crate. May be we should have a `tedge-common` crate or something like that for things like the command log, the workflow engine etc.

Reply:
> Changes like this makes me feel that we're bloating the `tedge-api` crate. May be we should have a `tedge-common` crate or something like that for things like the command log, the workflow engine etc.

Indeed an issue.

Looking closer, this `nix` crate is used in a single place: the private function `send_signal_to_stop_child`.
So an option is to:
- move the function `send_signal_to_stop_child` to the `tedge_utils` crate which already depends on `nix`
- later we will be in position to provide an implementation for non nix systems
- I would just try to find a better name as `stop_process`

For the workflow engine, a different approach has been chosen. `tedge_api` only defines workflow related structs, while all the workflow side-effects (as MQTT and persistence) are implemented by `tedge_agent`.

Reply:
Resolved as suggested.